src/misc.js

```diff
@@ -265,4 +265,19 @@ engine.singleton = function (coll, type) {
   throw new Error('Not supported type ' + type);
 };
 
+/**
+ * Checks whether a primitve value is present
+ */
+engine.hasValueFn = function(coll) {
+  if (coll.length === 1){
+    return [isPrimitive(coll[0].data)];
+  } else {
+    return [false];
+  }
+};
+
+function isPrimitive(data){
+  return data !== null && typeof data !== "object";
```

Comment:
`typeof null === 'object'` so we don't need the first check.

@plynchnlm I'm wondering if we also need to check the node type in the model.
This will affect performance. What do you think?
We can read the node type from `coll[0].path` when a model is specified.
The model can be specified by adding: `model: r4` to the test description in hasValue.yaml
The list of primitive types is here:
https://hl7.org/fhir/datatypes.html#primitive

Reply:
@plynchnlm If no model is specified, should this method work?

Reply:
I think if no FHIR model is specified, hasValue need not work, since it is a FHIR extension.  However, I suppose you could have two implementations, one slower one checking type when there isn't a model, and one faster one when the model is available.